src/unicode.cpp

```diff
@@ -60,22 +60,20 @@ ada_really_inline bool has_tabs_or_newline(
   }
   // fast path for long strings (expected to be common)
   size_t i = 0;
-  const uint8x16_t mask1 = vmovq_n_u8('\r');
-  const uint8x16_t mask2 = vmovq_n_u8('\n');
-  const uint8x16_t mask3 = vmovq_n_u8('\t');
+  static uint8_t rnt_array[16] = {1, 0,  0, 0, 0,  0, 0, 0,
```

Comment:
For read only algorithms, I tend to align data accesses.
That also allows handling of tails for free (aligned load for 16 bytes never crashes, there is a a way to convince ASAN to go away).

Here is some code in folly that does it, it's Appache-2, I don't know licenses, but you probably can use it.

https://github.com/facebook/folly/blob/dcdbd4c8d5f6511086b219dacb3e7aeb7889c662/folly/detail/SimdForEach.h#L55

https://github.com/facebook/folly/blob/dcdbd4c8d5f6511086b219dacb3e7aeb7889c662/folly/detail/SimdAnyOf.h#L76

Here is the same code in eve, it's Boost license. You can 100% use it, but it has eve's idiocincratisis. 
https://github.com/jfalcou/eve/blob/a2e2cf539e36e9a3326800194ad5206a8ef3f5b7/include/eve/module/algo/algo/for_each_iteration.hpp#L148

Reply:
I have not looked at the links. This being said, if the input is at least 16 bytes,  we already handle the tail efficiently. If the input is less than 16 bytes, then you will read beyond your buffer if you try to use SIMD. You can tell some sanitizers not to report the error, but to my knowledge, there is no way to tell every single bound checker (valgrind, default visual studio bound checker, etc.) one might use to ignore the issue.  And, it is unclear whether a SIMD algorithm over a string of less than 16 bytes is warranted. Scalar code can be faster on small inputs.